tracing-opentelemetry/src/tracer.rs

```diff
@@ -0,0 +1,96 @@
+use opentelemetry::{api, sdk};
+
+/// A protocol for OpenTelemetry [`Tracer`]s that are capable of producing
+/// sampled span contexts _before_ starting their associated spans.
+///
+/// This enables interoperability between `tracing` and `opentelemetry` by
+/// allowing otel trace ids to be associated _after_ a `tracing` span has been
+/// created. See the [`OpenTelemetrySpanExt`] in this crate for usage examples.
```

Comment:
I don't _entirely_ understand what this trait is doing...it would be nice if there was a little more explanation here.

Reply:
Can add more comments :)

Reply:
Clearer with new comments?

Reply:
Yup, seems much better! Thanks!